plugins/modules/gitlab_issue.py

```diff
@@ -0,0 +1,412 @@
+            self._module.exit_json(changed=True, msg="Successfully updated Issue '%s'." % issue["title"])
+
+        try:
+            return self.project.issues.update(issue.iid, options)
+        except gitlab.exceptions.GitlabUpdateError as e:
+            self._module.fail_json(msg="Failed to update Issue %s." % to_native(e))
+
+    '''
+    @param issue Issue object to evaluate
+    @param options New options to update Issue with
+    '''
+    def issue_has_changed(self, issue, options):
+        for key, value in options.items():
+            if value is not None:
+
+                if key == 'milestone_id':
+                    old_milestone = getattr(issue, 'milestone')['id'] if getattr(issue, 'milestone') else ""
+                    if options[key] != old_milestone:
+                        return True
+                elif key == 'assignee_ids':
+                    if options[key] != sorted([user["id"] for user in getattr(issue, 'assignees')]):
+                        return True
+
+                elif key == 'labels':
+                    if options[key] != sorted(getattr(issue, key)):
+                        return True
+
+                elif getattr(issue, key) != value:
+                    return True
+
+        return False
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(auth_argument_spec())
+    argument_spec.update(
+        assignee_ids=dict(type='list', elements='str', required=False),
+        description=dict(type='str', required=False),
+        description_path=dict(type='path', required=False),
+        issue_type=dict(type='str', default='issue', choices=["issue", "incident", "test_case"], required=False),
+        labels=dict(type='list', elements='str', required=False),
+        milestone_id=dict(type='str', required=False),
+        milestone_group_id=dict(type='str', required=False),
+        project=dict(type='str', required=True),
+        state=dict(type='str', default="present", choices=["absent", "present"]),
+        state_filter=dict(type='str', default="opened", choices=["opened", "closed"]),
+        title=dict(type='str', required=True),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['api_username', 'api_token'],
+            ['api_username', 'api_oauth_token'],
+            ['api_username', 'api_job_token'],
+            ['api_token', 'api_oauth_token'],
+            ['api_token', 'api_job_token'],
+            ['description', 'description_path'],
+        ],
+        required_together=[
+            ['api_username', 'api_password'],
+            ['milestone_id', 'milestone_group_id'],
+        ],
+        required_one_of=[
+            ['api_username', 'api_token', 'api_oauth_token', 'api_job_token']
+        ],
+        required_if=[
+            ['state', 'present', ['project', 'title'], True],
+            ['state', 'absent', ['project', 'title'], True],
+        ],
```

Comment:
Since `project` and `title` are required, are these lines really necessary ?

Reply:
Agreed, cleaned up. Thanks.